jestHelpers/ConsoleUsageReporter.js

```diff
@@ -0,0 +1,172 @@
+            .map(line => `  ${line}`)
+            .join('\n')
+
+          const formattedCallStack = !/^\s*(at|in) /m.test(formattedArgs)
+            ? red(
+                `\n\n${callStack
+                  .split('\n')
+                  .map(line => `  ${line}`)
+                  .join('\n')}`
+              )
+            : ''
+
+          return `console.${method}\n${reset(
+            formattedArgs
+          )}${formattedCallStack}`
+        })
+        .join('\n\n')
+    ;['error', 'info', 'log', 'warn'].forEach(method => {
+      global.console[method] = (...args) => {
+        const callStack = new Error().stack
+          .split('\n')
+          .slice(2)
+          .map(line => line.trim())
+          .join('\n')
+
+        if (consoleCalls.length === 0) {
+          ConsoleUsageReporter.recordConsoleUsedInCurrentTestFile()
+        }
+
+        if (testsRunning) {
+          consoleCalls.push({ args, callStack, method })
+        } else {
+          process.stderr.write(
+            red(`
+The console has been called outside a test which usually means you mishandled asynchronous actions.
+
+Here is what have been logged:
+
+${reset(formatConsoleCalls([{ args, callStack, method }]))}
+`)
+          )
+        }
+      }
+    })
+
+    beforeAll(() => {
+      testsRunning = true
+    })
+
+    beforeEach(() => {
+      consoleCalls = []
+    })
+
+    afterEach(() => {
+      if (consoleCalls.length > 0) {
+        throw new Error(
+          red(`\
+This test called the console which is forbidden.
+
+Here is what have been logged:
+
+${reset(formatConsoleCalls(consoleCalls))}
+
+If calling the console is normal in your test case, consider mocking the \
+console as is:
+
+  const consoleSpy = jest.spyOn(console, 'method').mockImplementation();
+  [...]
+  consoleSpy.mockRestore();
+`)
+        )
+      }
+    })
+
+    afterAll(() => {
+      testsRunning = false
+    })
+  }
+
+  constructor(globalConfig) {
+    this.globalConfig = globalConfig
+  }
+
+  onRunComplete() {
+    const isWatchModeEnabled =
+      this.globalConfig.watch || this.globalConfig.watchAll
+    const testFilesThatUsedConsole = ConsoleUsageReporter.getTestFilesThatUsedConsole()
+
+    ConsoleUsageReporter.deleteTemporaryFile()
+
+    if (testFilesThatUsedConsole.length > 0) {
+      const error = new Error(
+        red(
+          `\
+The following test files called the console which is forbidden:
+${testFilesThatUsedConsole.map(file => `- ${file}`).join('\n')}
+
+You should find more information in the report of the test that called the \
+console.
+
+We list all the test files there to allow you to find the console calls that \
+did not make any test fail (possibly because of async issues).
+`
+        )
+      )
+
+      if (isWatchModeEnabled) {
+        // Prevents to freeze watch mode
+        console.error(error)
+      } else {
+        throw error
+      }
+    }
+  }
+
+  onRunStart() {
+    ConsoleUsageReporter.deleteTemporaryFile()
+  }
+}
```

Comment:
I don't understand what this file really do 🤷‍♂️

Reply:
Whenever a console.error is thrown while running jest, the ConsoleUsageReporter will intercept it and make the test failing and indicating how to solve the issue by mocking the console for example.

The advantage of this, is to fight against breaking changes. Whenever a breaking changes occur while upgrading dependencies, the warning in the console - that indicates the breaking changes - will make the "unit test" jobs failing.

Reply:
I remove the file. This PR can be merged, I guess